packages/react/src/components/header/components/headerActionBarItem/HeaderActionBarItem.tsx

```diff
@@ -1,144 +1,180 @@
-import React, { cloneElement, forwardRef, ReactNode } from 'react';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useEffect, useRef, useState } from 'react';
 
 import classes from './HeaderActionBarItem.module.scss';
+import { HeaderActionBarItemButton, HeaderActionBarItemButtonProps } from './HeaderActionBarItemButton';
+import { useHeaderContext } from '../../HeaderContext';
 import classNames from '../../../../utils/classNames';
 
-type ButtonAttributes = JSX.IntrinsicElements['button'];
-
-export interface HeaderActionBarItemProps extends ButtonAttributes {
+export type HeaderActionBarItemProps = React.PropsWithChildren<{
+  /**
+   * Possibility to use a full-width version of the dropdown, for example in mobile use.
+   */
+  fullWidth?: boolean;
   /**
-   * Id of controlled dropdown menu.
+   * Label for the action bar item.
    */
-  ariaControls?: string;
+  label?: string | JSX.Element;
   /**
-   * Aria label for the item.
+   * Aria-label attribute for the dropdown button.
    */
   ariaLabel?: string;
+  /**
+   * Positions the label right side of the icon.
+   */
+  labelOnRight?: boolean;
   /**
    * Fix the item position to the right side of the action bar.
    */
   fixedRightPosition?: boolean;
   /**
-   * Icon element for the action bar item.
+   * Initials for avatar which replace icon.
    */
-  icon?: ReactNode;
+  avatar?: string | JSX.Element;
```

Comment:
ActionBarItem can now be a button, link, heading and also an avatar.

This multitasking complicates the code and typescript definitions.

For example the `icon` must be provided, but the avatar version does not use it. 

Maybe we should least separate the avatar version to its own component with own props?